src/sentry/search/snuba/backend.py

```diff
@@ -353,8 +362,12 @@ def do_search(project_id, environment_id, tags, start, end,
                 group_data[group_id] = defaultdict(list)
 
             dest = group_data[group_id]
-            for k, v in obj.items():
-                dest[k].append(v)
+            if len(required_aggregations) == 1:
```

Comment:
I think you mentioned this earlier today maybe but the different return type based on sequence length here is kind of weird.

Reply:
Yeah, talked to Hoff, so far this is the one weird place where we have to handle it, and it apparently makes all of TSDB a lot simpler. So I think I'm going to go with a comment here and we can revisit with either a Snuba option to not do this, or rollback the feature entirely -- if we run into it more.